pkg/controller/nodeipam/ipam/cidr_allocator.go

```diff
@@ -144,3 +161,44 @@ func listNodes(kubeClient clientset.Interface) (*v1.NodeList, error) {
 	}
 	return nodeList, nil
 }
+
+func listClusterCIDRs(kubeClient clientset.Interface) (*networkingv1alpha1.ClusterCIDRList, error) {
+	var clusterCIDRList *networkingv1alpha1.ClusterCIDRList
+	// We must poll because apiserver might not be up. This error causes
+	// controller manager to restart.
+	startTimestamp := time.Now()
+
+	// start with 2s, multiply the duration by 1.6 each step, 11 steps = 9.7 minutes
```

Comment:
probably want fewer steps (cap it at ~1-2 minutes)

Reply:
Wanted to align with the apeiserverstartupGracePeriod: https://github.com/kubernetes/kubernetes/blob/cdc60112a676010dbc0ba5650732386e2dedebd0/pkg/controller/nodeipam/ipam/cidr_allocator.go#L57-L58